pkg/service/aws/volume.go

```diff
@@ -160,11 +140,11 @@ func (svc awsController) DeleteVolume(ctx context.Context, req *proto.DeleteVolu
 	//calling aws sdk method to delete volume
 	//ec2.DeleteVolumeOutput doesn't contain anything
 	//hence not taking response
-	_, err = ec2Client.DeleteVolume(input)
+	_, err = ec2Client.DeleteVolumeWithContext(ctx, input)
```

Comment:
We should also wait till the volume is deleted else `ctx` will get canceled as soon as the request returns and `DeleteVolumeWithContext` might also get canceled.

Reply:
no, we would submit the call to aws and it will proceed to complete. Once the submitted context has no control on the request. You can cancel it immediately.